extensions/prototypes/labgraph_monitor/README.md

```diff
@@ -0,0 +1,66 @@
+# LabGraph Monitor
+
+LabGraph Monitor helps visualize the graph built. Inorder to run the application, you will need to first set it up using React using the below steps:
+
+First, incase you do not have npm installed, follow the below steps to get started:
+
+Download Node.js from [here](https://nodejs.org/en/download/) 
+
+Once it is installed, you can verify that node and npm got downloaded successfully by running the following: 
+
+```
+node -v
+```
+
+```
+npm -v
+```
+
+The above should display the version of node/npm that got installed 
+
+Example:
+
+```
+6.14.13
+```
+
+Now that you have npm installed, you can create a new React application using the terminal. Navigate to the folder where you want the application to be installed and run the below code to create a new React application. We are going to call the application "labgraph_monitor_extension" for the sake of this example but feel free to choose another name. 
+
+Note: npx is a package that comes with npm 5.2+ 
+```
+npx create-react-app labgraph_monitor_extension
+```
+
+Next, you will need to install dagre library and ReactFlow that help in layouting the graph. 
+Run the below code to install them
+```
+npm i dagre
+```
+
+```
+npm install --save react-flow-renderer
+```
+
+Once the application is created, we will need to change and add some files to get the LabGraph Monitor working. 
+
+First, you can validate that everything is going well so far by running the new sample React app in your browser. Navigate using the terminal to the labgraph_monitor_extension folder and run
+
+```
+npm start
+```
+
+This should open a sample React app in your localhost and display the React icon. 
+
+Now that we have the initial application set up, we can start adding the new files. 
+
+Add the App.js, helper.js and App.css files downloaded from [labgraph repo](https://github.com/facebookresearch/labgraph) found at .\labgraph\extensions\prototypes\labgraph_monitor\src to your labgraph_monitor_extension application. You need to add them in the ./labgraph_monitor_extension/src folder 
```

Comment:
Is there a way to simplify this process? Currently the user has to do quite a bit to get started with the application, and there are quite a few points of potential failure.

Reply:
As labgraph monitor uses React, the user needs to set up a new React application themselves using the steps before that and once done, they will need to add the files that make the monitor work (the three files mentioned above). I have initially added all the files needed to directly run the application from labgraph repo without creating a new React application themselves ,but since there were alot of dependencies installed with the React application, I have removed them and only committed these main files that they need to add. That's why I don't think if it can be simplified further since they need to add these files to get it to work.